### PR TITLE
Add and Refactor Diet

### LINKED ISSUE #15: Add meals tracker
As a healthy user, I want to keep track of my daily intake of calories for each meal and amount of protein, carb and fat consumed.

### PR DESCRIPTION
**Pull request by nihalzp**
Closes #15

### CODE REVIEW

File: src/main/java/athleticli/data/diet/DietGoal.java [hunk 1]
```diff
@@ -1,4 +1,126 @@
...
+     *
+     * @param currentValue The current value of the diet goal.
+     */
+    public void setCurrentValue(int currentValue) {
+        this.currentValue = currentValue;
+        if (!isGoalAchieved && currentValue >= targetValue) {
+            setIsGoalAchieved(true);
+        } else if (isGoalAchieved && currentValue < targetValue) {
+            setIsGoalAchieved(false);
+        }
```
Comment (skylee03) [inline src/main/java/athleticli/data/diet/DietGoal.java:96]:
```suggestion
        setIsGoalAchieved(currentValue >= targetValue);
```

File: src/test/java/athleticli/data/diet/DietGoalListTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @Test
+    void add_addOneGoal_expectSizeOne() {
+        dietGoals.add(proteinGoal);
+        assertEquals(1, dietGoals.size());
+    }
+
+    @Test
+    void remove_removeExistingGoal_expectSizeOne() {
+        dietGoals.add(proteinGoal);
+
```
Comment (skylee03) [inline src/test/java/athleticli/data/diet/DietGoalListTest.java:29]:
```suggestion
```

File: src/test/java/athleticli/data/diet/DietGoalListTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    void remove_removeFromZeroGoals_expectIndexOutOfRangeError() {
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            dietGoals.remove(0);
+        });
+    }
+
+    @Test
+    void get_addOneGoal_expectGetSameGoal() {
+        dietGoals.add(proteinGoal);
+
```
Comment (skylee03) [inline src/test/java/athleticli/data/diet/DietGoalListTest.java:44]:
```suggestion
```

File: src/test/java/athleticli/data/diet/DietGoalListTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        for (int i = 0; i < 10; i++) {
+            dietGoals.add(proteinGoal);
+        }
+        assertEquals(10, dietGoals.size());
+    }
+
+    @Test
+    void testToString_oneExistingGoal_expectCorrectFormat() {
+        dietGoals.add(proteinGoal);
+
```
Comment (skylee03) [inline src/test/java/athleticli/data/diet/DietGoalListTest.java:64]:
```suggestion
```

**Review by skylee03 — COMMENTED** [review]
LGTM! I suggested one change in `setCurrentValue()` to match the code logic of `setTargetValue()`. Additionally, I would also recommend removing some redundant blank lines.

**Comment by nihalzp** [comment]
Thanks for suggestions. I have amended them.

**Review by skylee03 — APPROVED** [review]
LGTM!

**Pull request merged**